### ISSUE TITLE
[bug] tauri-cli incorrectly parses application arguments

### ISSUE THREAD

**Issue by fpgaminer** [tauri-apps/tauri#5415]
### Describe the bug

Running `cargo tauri dev -- -- --myoption` causes tauri to pass `--no-default-features` to my application. That means `cargo build` doesn't get that flag, and it means my application crashes because `clap` isn't expecting the `--no-default-features` flag in my application.

This seems to be due to this code: https://github.com/tauri-apps/tauri/blob/4036e15f5af933bdc0d0913508b5103958afc143/tooling/cli/src/interface/rust.rs#L303

After that line the rest of the function splits up `args` into `args` and `run_args` by looking for `--`.  Since `--no-default-features` is `push`ed onto `args` beforehand, it will incorrectly end up in `run_args`.

Likely the code that pushes `--no-default-features` should either use an insert to put the flag at the beginning of `args`, or the push needs to occur _after_ `run_args` has been split off.

### Platform and versions

```
Environment
  › OS: Mac OS 12.6.0 X64
  › Node.js: 18.10.0
  › npm: 8.19.2
  › pnpm: Not installed!
  › yarn: Not installed!
  › rustup: 1.25.1
  › rustc: 1.64.0
  › cargo: 1.64.0
  › Rust toolchain: stable-aarch64-apple-darwin 

Packages
  › @tauri-apps/cli [NPM]: 1.1.1
  › @tauri-apps/api [NPM]: 1.1.0
  › tauri [RUST]: 1.1.1,
  › tauri-build [RUST]: 1.1.1,
  › tao [RUST]: 0.14.0,
  › wry [RUST]: 0.21.1,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../dist
  › devPath: http://localhost:1420/
  › framework: React

App directory structure
  ├─ dist
  ├─ node_modules
  ├─ public
  ├─ src-tauri
  ├─ .vscode
  └─ src

```

**Comment by e792a8** [comment]
This bug also causes the app ui not updating when frontend file changes. Cargo do not receive '--no-default-features', so the app is not connecting to frontend dev server but using old dist files.

**Comment by FabianLars** [comment]
@e792a8 This issue was fixed in the cli v1.2.0. Can you make sure your dependencies are up to date?

Also you can remove the default-features from the Cargo.toml file without any consequences (starting with v1.3 it won't be part of the templates anymore).

**Comment by e792a8** [comment]
@FabianLars I know... I have just encountered these problems, then I'm here to find it already fixed. Updating to cli v1.2 solved these. Thank u
